winch/codegen/src/codegen/call.rs

```diff
@@ -330,16 +333,28 @@ impl FnCall {
                 let result_bytes = sig.results_stack_size();
                 debug_assert!(sp.as_u32() >= stack_consumed + result_bytes);
                 let dst = SPOffset::from_u32(sp.as_u32() - stack_consumed);
-                masm.memmove(sp, dst, result_bytes);
+                masm.memmove(sp, dst, result_bytes, MemMoveDirection::LowToHigh);
             }
         };
 
         // Free the bytes consumed by the call.
         masm.free_stack(stack_consumed);
 
-        let mut results_data = ABIResultsData::wrap(sig.results.clone());
-        results_data.ret_area = ret_area;
+        if let Some(area) = ret_area {
+            if stack_consumed > 0 {
```

Comment:
Is it worth adding an assert here that the area is not `Uninit`?

Reply:
Good point, I added an assert!